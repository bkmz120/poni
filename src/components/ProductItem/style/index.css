.productItem {
  width: 200px;
  box-shadow: 0 0 11px 0 #dadada;
  padding: 10px;
  margin: 10px;
}

.label {
  color: #ccc;
  font-size: 13px;
  margin-bottom: 5px;
}

.value {
  color: #333;
  margin-bottom: 8px;
}

.isNew {
  background: #e27d7d;
  color: #fff;
  padding: 5px 10px;
  display: inline-block;
  font-size: 14px;
  font-weight: 700;
}
