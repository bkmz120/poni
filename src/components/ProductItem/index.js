import React from 'react'
import PropTypes from 'prop-types'

import { productShape } from '@shapes'
import style from './style/index.css'

export default class ProductItem extends React.PureComponent {
  render () {
    const {
      product
    } = this.props

    console.log('render')

    return <div className={style.productItem}>
      <div className={style.label}>
        Имя
      </div>
      <div className={style.value}>
        {product.name}
      </div>
      <div className={style.label}>
        Цвет
      </div>
      <div className={style.value}>
        {product.color}
      </div>
      <div className={style.label}>
        Вид
      </div>
      <div className={style.value}>
        {product.kind}
      </div>
      <div className={style.label}>
        Цена
      </div>
      <div className={style.value}>
        {product.price}
      </div>
      <div className={style.label}>
        Новый
      </div>
      <div className={style.value}>
        {product.is_new}
      </div>
    </div>
  }
}

ProductItem.propTypes = {
  product: PropTypes.shape(productShape).isRequired,
}
