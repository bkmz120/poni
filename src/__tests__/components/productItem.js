/* eslint-disable */

import ProductItem from '@components/ProductItem'

describe('ProductItem',()=>{
  const product = {
    'name': 'Абдула',
    'color': 'Синий',
    'kind': 'Аликорн',
    'price': 29.99,
    'is_new': true,
  }

  it('check snapshot', () => {
    const itemWrapper = shallow(
        <ProductItem
          product={product}
        />
    );
    expect(itemWrapper).toMatchSnapshot()
  })

  it('should contain name, color, kind, price, is_new', () => {
    const itemWrapper = shallow(
        <ProductItem
          product={product}
        />
    );

    expect(itemWrapper.text().includes(product.name)).toBe(true)
    expect(itemWrapper.text().includes(product.color)).toBe(true)
    expect(itemWrapper.text().includes(product.kind)).toBe(true)
    expect(itemWrapper.text().includes(product.price)).toBe(true)
    expect(itemWrapper.text().includes('Новинка!')).toBe(true)
  })
})

