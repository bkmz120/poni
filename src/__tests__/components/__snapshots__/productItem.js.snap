// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem check snapshot 1`] = `
<div
  className="productItem"
>
  <div
    className="label"
  >
    Имя
  </div>
  <div
    className="value"
  >
    Абдула
  </div>
  <div
    className="label"
  >
    Цвет
  </div>
  <div
    className="value"
  >
    Синий
  </div>
  <div
    className="label"
  >
    Вид
  </div>
  <div
    className="value"
  >
    Аликорн
  </div>
  <div
    className="label"
  >
    Цена
  </div>
  <div
    className="value"
  >
    29.99
  </div>
  <div
    className="isNew"
  >
    Новинка!
  </div>
</div>
`;
